pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,131 @@
+package agent
+
+import (
+	//corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+	APIVIP            string `json:"apiVIP"`
+	APIVIPDNSName     string `json:"apiVIPDNSName"`
+	IngressVIP        string `json:"ingressVIP"`
+	VIPDHCPAllocation bool   `json:"vipDHCPAllocation"`
+
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	Networking *Networking `json:"networking,omitempty"`
+
+	// Proxy defines the proxy settings for the cluster.
+	// If unset, the cluster will not be configured to use a proxy.
+	// +optional
+	Proxy *Proxy `json:"proxy,omitempty"`
+
+	UserManagedNetworking  bool   `json:"userManagedNetworking,omitempty"`
+	AdditionalNtpSource    string `json:"additionalNtpSource,omitempty"`
+	InstallConfigOverrides string `json:"installConfigOverrides,omitempty"`
+
+	SshPublicKey string `json:"sshPublicKey,omitempty"`
```

Comment:
Still a bit unsure what to do with this public key here. 

I assume this is used to place in the generated InstallConfig?

Hive has cd.Spec.Provisioning.SSHPrivateKeySecretRef. THis is *assumed* to match whatever was in the install-config you gave us, and only used to gather logs from hosts in the event of a failure.

In theory, with the private key we can get the public key.

```bash
ssh-keygen -y -f ~/.ssh/id_rsa > ~/.ssh/id_rsa.pub
```

But I'm not super eager to start shelling out. Tempted to not mandate giving us a private key and let assisted callers specify a public key only.

Reply:
i think requiring an ssh private key will be key for debugging info. so i think we should keep the private key and update the assisted installer code to extract public key before giving it to the installer..

Reply:
Keeping this here for now, flagged for possibly moving up to higher in Hive stack someday before we go stable.